File: 04.Simulation模拟/leetcode_1967_numOfStrings.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    int numOfStrings(vector<string>& patterns, string word) {
        int res = 0;
        for (auto& s : patterns){
            for (int i=0; i<word.size(); i++){
                if (word.substr(i, s.size()) == s){
                    res++;
                    break;
                }
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<string> patterns = {"a","abc","bc","d"};
    string word = "abc";
    cout << s.numOfStrings(patterns, word) << endl;
}

File: 04.Simulation模拟/leetcode_1967_numOfStrings.py
from typing import List

class Solution:
    def numOfStrings(self, patterns: List[str], word: str) -> int:
        return sum(i in word for i in patterns)

if __name__ == "__main__":
    s = Solution()
    patterns = ["a","abc","bc","d"]
    word = "abc"
    print(s.numOfStrings(patterns, word))
